### PR TITLE
security: Implement IPsecListSas API

### PR DESCRIPTION
**Pull request by mestery**
This commits implements the IPsecListSas API for both client and server.
This is tested by the existing code, which runs the client, in turn
calling the new API and printing out results.

Related to #320

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: security/server/ipsec_vici.go [hunk 1]
```diff
@@ -105,6 +105,74 @@ type rekey_connection struct {
 	Reauth     bool   `vici:"reauth"`
 }
 
+type list_sas struct {
+	Ike        string `vici:"ike"`
+	IkeId      string `vici:"ike"`
```
Comment (seroyer) [inline security/server/ipsec_vici.go:110]: Probably doesn't matter, but should this be `vici:"ike-id"` instead?

File: security/server/ipsec_vici.go [hunk 2]
```diff
@@ -105,6 +105,74 @@ type rekey_connection struct {
...
+	Child      string `vici:"child"`
+	ChildId    string `vici:"child"`
```
Comment (seroyer) [inline security/server/ipsec_vici.go:112]: Similarly, should this be `vici:"child-id"`?

File: security/server/ipsec_vici.go [hunk 3]
```diff
@@ -105,6 +105,74 @@ type rekey_connection struct {
...
+	NatAny        string                   `vici:"nat-any"`
+	IfIdIn        string                   `vici:"if-id-in"`
+	IfIdOut       string                   `vici:"if-id-out"`
+	EncrAlg       string                   `vici:"encr-alg"`
+	EncrKeysize   string                   `vici:"encr-keysize"`
+	IntegAlg      string                   `vici:"integ-alg"`
+	IntegKeysize  string                   `vici:"integ-keysize"`
+	PrfAlg        string                   `vici:"prf-alg"`
+	DhGroup       string                   `vici:"dh-group"`
+	Ppk           string                   `vici:"ppk"`
```
Comment (seroyer) [inline security/server/ipsec_vici.go:164]: I don't see this here, is there somewhere else I should be looking?  https://github.com/strongswan/strongswan/blob/master/src/libcharon/plugins/vici/README.md#list-sa

**Review by seroyer — APPROVED** [review]
I don't fully understand this, but it looks reasonable.

File: security/server/ipsec_vici.go (hunk 1 above)
Comment (mestery) [reply security/server/ipsec_vici.go:110]: Good catch Steve!

File: security/server/ipsec_vici.go (hunk 2 above)
Comment (mestery) [reply security/server/ipsec_vici.go:112]: Another nice find!

File: security/server/ipsec_vici.go (hunk 3 above)
Comment (mestery) [reply security/server/ipsec_vici.go:164]: Yes, the source code is our friend: https://github.com/strongswan/strongswan/blob/master/src/libcharon/plugins/vici/vici_query.c#L497

Which makes me think I should re-examine the connection stuff, and will do that next. Thanks again for the review!

**Comment by glimchb** [comment]
Excellent work
There is some code dup so after we finish with basic reference implementation maybe we can look how to reduce it...

**Pull request merged**